packages/next-intl/src/middleware/utils.tsx

```diff
@@ -86,7 +88,9 @@ export function getKnownLocaleFromPathname<Locales extends AllLocales>(
   locales: Locales
 ): Locales[number] | undefined {
   const pathLocaleCandidate = getLocaleFromPathname(pathname);
-  const pathLocale = locales.includes(pathLocaleCandidate)
+  const pathLocale = locales.find(
+    (locale) => locale.toLowerCase() === pathLocaleCandidate.toLowerCase()
+  )
```

Comment:
The value returned from this function is assigned to a variable here:

https://github.com/amannn/next-intl/blob/5cc264ddd769a57ea09440f25c26325c142d0c25/packages/next-intl/src/middleware/middleware.tsx#L137

… and since it doesn't equal the detected locale at 

https://github.com/amannn/next-intl/blob/5cc264ddd769a57ea09440f25c26325c142d0c25/packages/next-intl/src/middleware/middleware.tsx#L219

… a redirect is invoked:

https://github.com/amannn/next-intl/blob/5cc264ddd769a57ea09440f25c26325c142d0c25/packages/next-intl/src/middleware/middleware.tsx#L246

That is correct.

However, the `locale` that is returned from `resolveLocale` here:

https://github.com/amannn/next-intl/blob/5cc264ddd769a57ea09440f25c26325c142d0c25/packages/next-intl/src/middleware/middleware.tsx#L47

… doesn't pick up the prefix since it used to be case senstive.

Since we redirect based on `locale` (and not `pathLocale`), I noticed that that request like `createMockRequest('/EN', 'de')` in the test would redirect to `/de`.

I've added a small fix in https://github.com/amannn/next-intl/pull/861/commits/f05395bfa86c94bda0c03666180b1417db78c585 that uses the locale from the pathname instead in this case—hope this is ok for you!